### PR TITLE
added option to force https for images only

### PR DESCRIPTION
**Pull request by chrirauch**
I'm in the same boat as some others users that would like to only rewrite channel logos and image urls without rewriting m3u urls as mentioned in the following issues...

https://github.com/Threadfin/Threadfin/issues/304
https://github.com/Threadfin/Threadfin/issues/459

I added an option "Force HTTPS for images only" which basically only rewrites the image urls to https.
image

I think this would be useful for running Threadfin behind a reverse proxy where only port 443 is forwarded to serve cached channel logos and images. In my case my IPTV Provider does not serve those over https and i want to cache them to make plex happy. 

Sorry if the code is not clean, i never worked with go and probably did some stuff that is not considered best practice. But in my tests it is working as i expect it and i think by adding an additional option, it shouldn't break any existing functionality. 

If "Force HTTPS" is true, everything should work the same as it was before and it overrides the "Force HTTPS for images only" option.

Please have a look if this is something that would work for you, or let me know if i can/should adjust something

### CODE REVIEW

File: src/m3u.go [hunk 1]
```diff
@@ -256,8 +256,12 @@ func buildM3U(groups []string) (m3u string, err error) {
 		}
 
 		logo := ""
+		ForceHttpsForImages := Settings.ForceHttps
+		if(ForceHttpsForImages == false){
```
Comment (joshjryan) [inline src/m3u.go:260]:
```suggestion
		if !ForceHttpsForImages {
```

File: src/xepg.go [hunk 2]
```diff
@@ -50,6 +51,10 @@ func buildXEPG(background bool) {
 	if System.ScanInProgress == 1 {
 		return
 	}
+	ForceHttpsForImages = Settings.ForceHttps
+	if(ForceHttpsForImages == false){
```
Comment (joshjryan) [inline src/xepg.go:55]:
```suggestion
	if !ForceHttpsForImages {
```

**Comment by chrirauch** [comment]
@Fyb3roptik I don't understand your comment. It will of course use 443 for image urls if the new option "Force HTTPS for images only" is checked and it will use 443 for all urls if the legacy option "Force HTTPs" is checked. It's not like browsers are the target here. Plex happily plays streams over a http connection and displays logos and images if they are served over https. In a Browser sure, you cannot mix http and https, but image urls are the only urls processed by threadfin that end up in browser anyway.

In my case my provider does not support https at all and if i activate the existing "force https" option, plex fails to play any streams. I mean if i activate a buffer it works, but this adds some additional latency when switching channels in plex. 

In my case threadfin runs on the same machine as plex. On another machine, a reverse proxy handles ssl connections to threadfin to serve the channel logos and images.

I don't know how i could get a setup like that to work with a provider that does not support https. Also, the pr is really simple and really shouldn't break any existing setups.

**Comment by Fyb3roptik** [comment]
Again this goes against SSL Certificate standards. So no. I don't care that Plex CAN do it. You shouldn't be doing that as it risks compromising the SSL certificate

**Comment by Serph91P** [comment]
> Again this goes against SSL Certificate standards. So no. I don't care that Plex CAN do it. You shouldn't be doing that as it risks compromising the SSL certificate

Ok so no https for all users with a http m3u got it. Since an option for only using https for images goes against ssl certificate standards :)

to explain I get this after using https
```
2024-12-19 09:13:31 [Threadfin] FFMPEG log:             [tcp @ 0x5634b188c840] Connection to tcp://iptv.domain.tld:443 failed: Connection refused

2024-12-19 09:13:31 [Threadfin] [ERROR] FFMPEG error (Streaming was stopped by third party transcoder (FFmpeg / VLC)) - EC: 1204

2024-12-19 09:13:31 [Threadfin] FFMPEG log:             https://iptv.domain.tld:443/play/whatever/ts?: Connection refused
```

Since everything gets rewritten to https ;)

**Comment by Serph91P** [comment]
@Fyb3roptik maybe you can explain why.

**Comment by Serph91P** [comment]
> > Again this goes against SSL Certificate standards. So no. I don't care that Plex CAN do it. You shouldn't be doing that as it risks compromising the SSL certificate
> 
> Ok so no https for all users with a http m3u got it. Since an option for only using https for images goes against ssl certificate standards :)
> 
> to explain I get this after using https
> 
> ```
> 2024-12-19 09:13:31 [Threadfin] FFMPEG log:             [tcp @ 0x5634b188c840] Connection to tcp://iptv.domain.tld:443 failed: Connection refused
> 
> 2024-12-19 09:13:31 [Threadfin] [ERROR] FFMPEG error (Streaming was stopped by third party transcoder (FFmpeg / VLC)) - EC: 1204
> 
> 2024-12-19 09:13:31 [Threadfin] FFMPEG log:             https://iptv.domain.tld:443/play/whatever/ts?: Connection refused
> ```
> 
> Since everything gets rewritten to https ;)

just found out that if you force http for ffmpeg it still adds the port 443.

```

2024-12-19 10:29:48 [Threadfin] FFMPEG log:             [tcp @ 0x55bc48f27240] Connection to tcp://iptv.domain.tld:443 failed: Connection refused

2024-12-19 10:29:48 [Threadfin] [ERROR] FFMPEG error (Streaming was stopped by third party transcoder (FFmpeg / VLC)) - EC: 1204

2024-12-19 10:29:48 [Threadfin] FFMPEG log:             http://iptv.domain.tld:443/play/whatever/ts?: Connection refused
```

**Pull request closed without merging**